src/NLog/Logger.cs

```diff
@@ -95,14 +96,35 @@ public bool IsEnabled(LogLevel level)
             return this.GetTargetsForLevel(level) != null;
         }
 
+#if NET4_5 
         /// <summary>
         /// Writes the specified diagnostic message.
         /// </summary>
         /// <param name="logEvent">Log event.</param>
-        public void Log(LogEventInfo logEvent)
+        /// <param name="memberName">Method or property name of the caller.</param>
+        /// <param name="sourceFilePath">Full path of the source file that contains the caller.</param>
+        /// <param name="sourceLineNumber">Line number in the source file at which the method is called.</param>
+#else   
+        /// <summary>
+        /// Writes the specified diagnostic message.
+        /// </summary>
+        /// <param name="logEvent">Log event.</param>
+#endif
+        public void Log(LogEventInfo logEvent
+#if NET4_5
+           , [CallerMemberName] string memberName = ""
+           , [CallerFilePath] string sourceFilePath = ""
+           , [CallerLineNumber] int sourceLineNumber = 0
+#endif
+            )
         {
             if (this.IsEnabled(logEvent.Level))
             {
+#if NET4_5
+                logEvent.CallerMethodName = memberName;
+                logEvent.CallerSourceFilePath = sourceFilePath;
+                logEvent.CallerLineNumber = sourceLineNumber;
+#endif
```

Comment:
U'm not sure if we should capture this for **every** LogEvent.  The `sourceFilePath` can be pretty long. For example, this can be easily 100 chars. If we printing just a small message (say 10 chars)  without the need of the caller info, than the logevent is 10 times bigger then expected.  

What do you think?